{
  "uid" : "ef94189011acb3cc",
  "name" : "Verify presence and visibility of the Header for an authorized user",
  "fullName" : "tests.header_test.TestAuthHeaderPageStructure#test_hpa_01_01_verify_auth_header_presence_and_visibility",
  "historyId" : "06e714c13e1f9312a2229989ffa111bd",
  "time" : {
    "start" : 1728610975026,
    "stop" : 1728610975026,
    "duration" : 0
  },
  "status" : "broken",
  "statusMessage" : "selenium.common.exceptions.WebDriverException: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: chrome=129.0.6668.89)\nStacktrace:\n#0 0x55846b28602a <unknown>\n#1 0x55846af6c5e0 <unknown>\n#2 0x55846af647eb <unknown>\n#3 0x55846af54a09 <unknown>\n#4 0x55846af56744 <unknown>\n#5 0x55846af54cbd <unknown>\n#6 0x55846af5452f <unknown>\n#7 0x55846af54422 <unknown>\n#8 0x55846af52493 <unknown>\n#9 0x55846af52ada <unknown>\n#10 0x55846af6ef41 <unknown>\n#11 0x55846b0008c5 <unknown>\n#12 0x55846afe0b22 <unknown>\n#13 0x55846afffd7d <unknown>\n#14 0x55846afe08c3 <unknown>\n#15 0x55846afae6b3 <unknown>\n#16 0x55846afaf68e <unknown>\n#17 0x55846b250a2b <unknown>\n#18 0x55846b2549b1 <unknown>\n#19 0x55846b23d225 <unknown>\n#20 0x55846b255532 <unknown>\n#21 0x55846b22238f <unknown>\n#22 0x55846b274f28 <unknown>\n#23 0x55846b2750f3 <unknown>\n#24 0x55846b284e7c <unknown>\n#25 0x7fba05094ac3 <unknown>",
  "statusTrace" : "driver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"d334a1ec7e9f80039d9e7900e7f1d8e1\")>\n\n    @pytest.fixture()\n    @allure.step(f'Open page: {MainPageLinks.URL_MAIN_PAGE}')\n    def main_page_open(driver):\n>       driver.get(MainPageLinks.URL_MAIN_PAGE)\n\ntests/conftest.py:22: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py:353: in get\n    self.execute(Command.GET, {\"url\": url})\n/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py:344: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x7f3bf8a5a9c0>\nresponse = {'status': 500, 'value': '{\"value\":{\"error\":\"unknown error\",\"message\":\"unknown error: net::ERR_CONNECTION_REFUSED\\\\n  ...\\\\n#23 0x55846b2750f3 \\\\u003Cunknown>\\\\n#24 0x55846b284e7c \\\\u003Cunknown>\\\\n#25 0x7fba05094ac3 \\\\u003Cunknown>\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.WebDriverException: Message: unknown error: net::ERR_CONNECTION_REFUSED\nE         (Session info: chrome=129.0.6668.89)\nE       Stacktrace:\nE       #0 0x55846b28602a <unknown>\nE       #1 0x55846af6c5e0 <unknown>\nE       #2 0x55846af647eb <unknown>\nE       #3 0x55846af54a09 <unknown>\nE       #4 0x55846af56744 <unknown>\nE       #5 0x55846af54cbd <unknown>\nE       #6 0x55846af5452f <unknown>\nE       #7 0x55846af54422 <unknown>\nE       #8 0x55846af52493 <unknown>\nE       #9 0x55846af52ada <unknown>\nE       #10 0x55846af6ef41 <unknown>\nE       #11 0x55846b0008c5 <unknown>\nE       #12 0x55846afe0b22 <unknown>\nE       #13 0x55846afffd7d <unknown>\nE       #14 0x55846afe08c3 <unknown>\nE       #15 0x55846afae6b3 <unknown>\nE       #16 0x55846afaf68e <unknown>\nE       #17 0x55846b250a2b <unknown>\nE       #18 0x55846b2549b1 <unknown>\nE       #19 0x55846b23d225 <unknown>\nE       #20 0x55846b255532 <unknown>\nE       #21 0x55846b22238f <unknown>\nE       #22 0x55846b274f28 <unknown>\nE       #23 0x55846b2750f3 <unknown>\nE       #24 0x55846b284e7c <unknown>\nE       #25 0x7fba05094ac3 <unknown>\n\n/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/selenium/webdriver/remote/errorhandler.py:229: WebDriverException",
  "flaky" : false,
  "newFailed" : false,
  "beforeStages" : [ {
    "name" : "driver",
    "time" : {
      "start" : 1728610975026,
      "stop" : 1728610975794,
      "duration" : 768
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  }, {
    "name" : "main_page_open",
    "time" : {
      "start" : 1728610975794,
      "stop" : 1728610976216,
      "duration" : 422
    },
    "status" : "broken",
    "statusMessage" : "selenium.common.exceptions.WebDriverException: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: chrome=129.0.6668.89)\nStacktrace:\n#0 0x55846b28602a <unknown>\n#1 0x55846af6c5e0 <unknown>\n#2 0x55846af647eb <unknown>\n#3 0x55846af54a09 <unknown>\n#4 0x55846af56744 <unknown>\n#5 0x55846af54cbd <unknown>\n#6 0x55846af5452f <unknown>\n#7 0x55846af54422 <unknown>\n#8 0x55846af52493 <unknown>\n#9 0x55846af52ada <unknown>\n#10 0x55846af6ef41 <unknown>\n#11 0x55846b0008c5 <unknown>\n#12 0x55846afe0b22 <unknown>\n#13 0x55846afffd7d <unknown>\n#14 0x55846afe08c3 <unknown>\n#15 0x55846afae6b3 <unknown>\n#16 0x55846afaf68e <unknown>\n#17 0x55846b250a2b <unknown>\n#18 0x55846b2549b1 <unknown>\n#19 0x55846b23d225 <unknown>\n#20 0x55846b255532 <unknown>\n#21 0x55846b22238f <unknown>\n#22 0x55846b274f28 <unknown>\n#23 0x55846b2750f3 <unknown>\n#24 0x55846b284e7c <unknown>\n#25 0x7fba05094ac3 <unknown>\n\n",
    "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/pluggy/_callers.py\", line 103, in _multicall\n    res = hook_impl.function(*args)\n          ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/_pytest/fixtures.py\", line 1123, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/_pytest/fixtures.py\", line 902, in call_fixture_func\n    fixture_result = fixturefunc(**kwargs)\n                     ^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/auto-tests-python/auto-tests-python/tests/conftest.py\", line 22, in main_page_open\n    driver.get(MainPageLinks.URL_MAIN_PAGE)\n  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py\", line 353, in get\n    self.execute(Command.GET, {\"url\": url})\n  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py\", line 344, in execute\n    self.error_handler.check_response(response)\n  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n",
    "steps" : [ {
      "name" : "Open page: https://brainup.site/",
      "time" : {
        "start" : 1728610975795,
        "stop" : 1728610976216,
        "duration" : 421
      },
      "status" : "broken",
      "statusMessage" : "selenium.common.exceptions.WebDriverException: Message: unknown error: net::ERR_CONNECTION_REFUSED\n  (Session info: chrome=129.0.6668.89)\nStacktrace:\n#0 0x55846b28602a <unknown>\n#1 0x55846af6c5e0 <unknown>\n#2 0x55846af647eb <unknown>\n#3 0x55846af54a09 <unknown>\n#4 0x55846af56744 <unknown>\n#5 0x55846af54cbd <unknown>\n#6 0x55846af5452f <unknown>\n#7 0x55846af54422 <unknown>\n#8 0x55846af52493 <unknown>\n#9 0x55846af52ada <unknown>\n#10 0x55846af6ef41 <unknown>\n#11 0x55846b0008c5 <unknown>\n#12 0x55846afe0b22 <unknown>\n#13 0x55846afffd7d <unknown>\n#14 0x55846afe08c3 <unknown>\n#15 0x55846afae6b3 <unknown>\n#16 0x55846afaf68e <unknown>\n#17 0x55846b250a2b <unknown>\n#18 0x55846b2549b1 <unknown>\n#19 0x55846b23d225 <unknown>\n#20 0x55846b255532 <unknown>\n#21 0x55846b22238f <unknown>\n#22 0x55846b274f28 <unknown>\n#23 0x55846b2750f3 <unknown>\n#24 0x55846b284e7c <unknown>\n#25 0x7fba05094ac3 <unknown>\n\n",
      "statusTrace" : "  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/auto-tests-python/auto-tests-python/tests/conftest.py\", line 22, in main_page_open\n    driver.get(MainPageLinks.URL_MAIN_PAGE)\n  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py\", line 353, in get\n    self.execute(Command.GET, {\"url\": url})\n  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py\", line 344, in execute\n    self.error_handler.check_response(response)\n  File \"/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n",
      "steps" : [ ],
      "attachments" : [ ],
      "parameters" : [ {
        "name" : "driver",
        "value" : "<selenium.webdriver.chrome.webdriver.WebDriver (session=\"d334a1ec7e9f80039d9e7900e7f1d8e1\")>"
      } ],
      "stepsCount" : 0,
      "attachmentsCount" : 0,
      "shouldDisplayMessage" : true,
      "hasContent" : true
    } ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 1,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : true
  }, {
    "name" : "clear_allure_results_folder",
    "time" : {
      "start" : 1728610900918,
      "stop" : 1728610900918,
      "duration" : 0
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "afterStages" : [ {
    "name" : "driver::0",
    "time" : {
      "start" : 1728610976250,
      "stop" : 1728610976312,
      "duration" : 62
    },
    "status" : "passed",
    "steps" : [ ],
    "attachments" : [ ],
    "parameters" : [ ],
    "stepsCount" : 0,
    "attachmentsCount" : 0,
    "shouldDisplayMessage" : false,
    "hasContent" : false
  } ],
  "labels" : [ {
    "name" : "epic",
    "value" : "Test Header"
  }, {
    "name" : "parentSuite",
    "value" : "tests"
  }, {
    "name" : "suite",
    "value" : "header_test"
  }, {
    "name" : "subSuite",
    "value" : "TestHeaderPage"
  }, {
    "name" : "host",
    "value" : "fv-az740-570"
  }, {
    "name" : "thread",
    "value" : "2375-MainThread"
  }, {
    "name" : "framework",
    "value" : "pytest"
  }, {
    "name" : "language",
    "value" : "cpython3"
  }, {
    "name" : "package",
    "value" : "tests.header_test"
  }, {
    "name" : "resultFormat",
    "value" : "allure2"
  } ],
  "parameters" : [ ],
  "links" : [ ],
  "hidden" : false,
  "retry" : false,
  "extra" : {
    "severity" : "normal",
    "retries" : [ ],
    "categories" : [ {
      "name" : "Test defects",
      "matchedStatuses" : [ ],
      "flaky" : false
    } ],
    "history" : {
      "statistic" : {
        "failed" : 0,
        "broken" : 2,
        "skipped" : 0,
        "passed" : 22,
        "unknown" : 0,
        "total" : 24
      },
      "items" : [ {
        "uid" : "bc64bd4c7e8a5080",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/58//#testresult/bc64bd4c7e8a5080",
        "status" : "passed",
        "time" : {
          "start" : 1728524814613,
          "stop" : 1728524814641,
          "duration" : 28
        }
      }, {
        "uid" : "b48cefa373cc0903",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/57//#testresult/b48cefa373cc0903",
        "status" : "passed",
        "time" : {
          "start" : 1728438489251,
          "stop" : 1728438489277,
          "duration" : 26
        }
      }, {
        "uid" : "4406ff7d22efdf2",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/56//#testresult/4406ff7d22efdf2",
        "status" : "passed",
        "time" : {
          "start" : 1728352061895,
          "stop" : 1728352061923,
          "duration" : 28
        }
      }, {
        "uid" : "dabb905ce3ad4f06",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/55//#testresult/dabb905ce3ad4f06",
        "status" : "broken",
        "statusDetails" : "selenium.common.exceptions.TimeoutException: Message: Can't find element by locator https://brainup.site/groups",
        "time" : {
          "start" : 1728266164950,
          "stop" : 1728266164950,
          "duration" : 0
        }
      }, {
        "uid" : "50fe3031fc27be04",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/54//#testresult/50fe3031fc27be04",
        "status" : "passed",
        "time" : {
          "start" : 1728179607716,
          "stop" : 1728179607742,
          "duration" : 26
        }
      }, {
        "uid" : "7e16fe0ac56e18f2",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/53//#testresult/7e16fe0ac56e18f2",
        "status" : "passed",
        "time" : {
          "start" : 1728092819645,
          "stop" : 1728092819671,
          "duration" : 26
        }
      }, {
        "uid" : "60f96a341c8dead1",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/52//#testresult/60f96a341c8dead1",
        "status" : "passed",
        "time" : {
          "start" : 1728006522868,
          "stop" : 1728006522895,
          "duration" : 27
        }
      }, {
        "uid" : "ac803a99558214ea",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/51//#testresult/ac803a99558214ea",
        "status" : "passed",
        "time" : {
          "start" : 1727920153210,
          "stop" : 1727920153239,
          "duration" : 29
        }
      }, {
        "uid" : "7c50e855fe6a3772",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/50//#testresult/7c50e855fe6a3772",
        "status" : "passed",
        "time" : {
          "start" : 1727833606469,
          "stop" : 1727833606505,
          "duration" : 36
        }
      }, {
        "uid" : "6e062a9e4cc5ab53",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/49//#testresult/6e062a9e4cc5ab53",
        "status" : "passed",
        "time" : {
          "start" : 1727747744156,
          "stop" : 1727747744185,
          "duration" : 29
        }
      }, {
        "uid" : "4b7f228f64f52e6",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/48//#testresult/4b7f228f64f52e6",
        "status" : "passed",
        "time" : {
          "start" : 1727661028722,
          "stop" : 1727661028749,
          "duration" : 27
        }
      }, {
        "uid" : "d594252a16833d71",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/47//#testresult/d594252a16833d71",
        "status" : "passed",
        "time" : {
          "start" : 1727574846582,
          "stop" : 1727574846609,
          "duration" : 27
        }
      }, {
        "uid" : "3d72754b742ebe21",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/46//#testresult/3d72754b742ebe21",
        "status" : "passed",
        "time" : {
          "start" : 1727488078081,
          "stop" : 1727488078114,
          "duration" : 33
        }
      }, {
        "uid" : "16a4d16ec78e1d38",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/45//#testresult/16a4d16ec78e1d38",
        "status" : "passed",
        "time" : {
          "start" : 1727401772459,
          "stop" : 1727401772492,
          "duration" : 33
        }
      }, {
        "uid" : "6a899aa820a5e47",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/44//#testresult/6a899aa820a5e47",
        "status" : "passed",
        "time" : {
          "start" : 1727315258779,
          "stop" : 1727315258808,
          "duration" : 29
        }
      }, {
        "uid" : "3916b59013d6895b",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/43//#testresult/3916b59013d6895b",
        "status" : "passed",
        "time" : {
          "start" : 1727228917710,
          "stop" : 1727228917736,
          "duration" : 26
        }
      }, {
        "uid" : "f0594a41c34a577d",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/42//#testresult/f0594a41c34a577d",
        "status" : "passed",
        "time" : {
          "start" : 1727142466165,
          "stop" : 1727142466192,
          "duration" : 27
        }
      }, {
        "uid" : "92938ea017f09e6f",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/41//#testresult/92938ea017f09e6f",
        "status" : "passed",
        "time" : {
          "start" : 1727055998779,
          "stop" : 1727055998804,
          "duration" : 25
        }
      }, {
        "uid" : "acb6af660a416346",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/40//#testresult/acb6af660a416346",
        "status" : "passed",
        "time" : {
          "start" : 1726969927169,
          "stop" : 1726969927196,
          "duration" : 27
        }
      }, {
        "uid" : "29bb410c9fadeecc",
        "reportUrl" : "https://lambotik.github.io/auto-tests-python/39//#testresult/29bb410c9fadeecc",
        "status" : "passed",
        "time" : {
          "start" : 1726882929285,
          "stop" : 1726882929320,
          "duration" : 35
        }
      } ]
    },
    "tags" : [ ]
  },
  "source" : "ef94189011acb3cc.json",
  "parameterValues" : [ ]
}